components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/workflow/CancelPendingApplicationApprovalWorkflowExecutor.java

```diff
@@ -0,0 +1,275 @@
+        }
+    }
+
+    /**
+     * Removes an application's pending subscription processes from BPS
+     *
+     * @param application Application which has the subscriptions to be removed
+     * @throws WorkflowException
+     */
+    private void removeSubscriptionProcessesByApplication(Application application) throws WorkflowException {
+        ApiMgtDAO apiMgtDAO = new ApiMgtDAO();
+        CancelPendingSubscriptionWorkflowExecutor cancelPendingSubscriptionExecutor =
+                (CancelPendingSubscriptionWorkflowExecutor) WorkflowExecutorFactory.getInstance().
+                        getWorkflowExecutor(WorkflowConstants.WF_TYPE_AM_SUBSCRIPTION_DELETION);
+        String subContentType = cancelPendingSubscriptionExecutor.getContentType();
+        String subUsername = cancelPendingSubscriptionExecutor.getUsername();
+        String subPassword = cancelPendingSubscriptionExecutor.getPassword();
+        String subServiceEndPoint = cancelPendingSubscriptionExecutor.getServiceEndpoint();
+        try {
+            Set<Integer> pendingSubscriptions = apiMgtDAO.getPendingSubscriptionsByApplicationId(application.getId());
+            Map<String, String> args = new HashMap<String, String>();
+            args.put("action", "http://workflow.subscription.apimgt.carbon.wso2.org/cancel");
+            args.put("username", subUsername);
+            args.put("password", subPassword);
+            args.put("serviceEndpoint", subServiceEndPoint);
+            args.put("contentType", subContentType);
+            ServiceClient client = getClient(args);
+
+            for (int subscription : pendingSubscriptions) {
+                String workflowExtRef = null;
+                try {
+                    workflowExtRef = apiMgtDAO.getExternalWorkflowReferenceForSubscription(subscription);
+                    String payload = "  <wor:CancelSubscriptionApprovalWorkflowProcessRequest " +
+                            "           xmlns:wor=\"http://workflow.subscription.apimgt.carbon.wso2.org\">\n" +
+                            "               <wor:workflowExtRef>" + workflowExtRef + "</wor:workflowExtRef>\n" +
+                            "           </wor:CancelSubscriptionApprovalWorkflowProcessRequest>";
+                    client.fireAndForget(AXIOMUtil.stringToOM(payload));
+                } catch (AxisFault axisFault) {
+                    log.error("Error sending out message", axisFault);
+                    throw new WorkflowException("Error sending out message", axisFault);
+                } catch (XMLStreamException e) {
+                    log.error("Error converting String to OMElement", e);
+                    throw new WorkflowException("Error converting String to OMElement", e);
+                }
+            }
+        } catch (APIManagementException e) {
+            log.error(e.getMessage(), e);
+            throw new WorkflowException(e.getMessage(), e);
+        } catch (AxisFault axisFault) {
```

Comment:
Axis fault seems occurred in two places but exception message same. Can we place specific error message?